src/crimson/os/seastore/cache.cc

```diff
@@ -245,6 +246,77 @@ void Cache::register_metrics()
       ),
     }
   );
+
+  /*
+   * cache_query: cache_access and cache_hit
+   */
+  auto register_cache_access =
+    [this, &labels_by_src, &labels_by_ext](src_t src, extent_types_t ext) {
+      auto m_key = std::make_pair(src, ext);
+      stats.cache_query[m_key] = {0, 0};
+      std::ostringstream oss_labels;
+      oss_labels << "(src=" << src << ", ext=" << ext << ")";
+      auto str_labels = oss_labels.str();
+      {
+        metrics.add_group(
+          "cache",
+          {
+            sm::make_counter(
+              "cache_access",
+              stats.cache_query.find(m_key)->second.access,
+              sm::description("total number of cache accesses " + str_labels),
+              {labels_by_src.find(src)->second,
+               labels_by_ext.find(ext)->second}
+            ),
+            sm::make_counter(
+              "cache_hit",
+              stats.cache_query.find(m_key)->second.hit,
+              sm::description("total number of cache hits " + str_labels),
+              {labels_by_src.find(src)->second,
+               labels_by_ext.find(ext)->second}
+            ),
+          }
+        );
+      }
+    };
+
+  for (auto& [src, label] : labels_by_src) {
+    for (auto& [ext, _label] : labels_by_ext) {
+      if (ext != extent_types_t::RETIRED_PLACEHOLDER) {
+        register_cache_access(src, ext);
```

Comment:
why not just pass `label` and `label_` (which deserves a better name, by the way) to  `register_cache_access()`?

Reply:
fixed.